### PR TITLE
fixed to possible use ip source as table name sql, and add timestamps…

### PR DESCRIPTION
**Pull request by 143981**
… for events.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by bazsi** [comment]
Ok to test

File: modules/afsql/afsql.c [hunk 1]
```diff
@@ -557,7 +557,7 @@ _table_create(AFSqlDestDriver *self, const gchar *table)
       return FALSE;
     }
 
-  g_string_printf(query_string, "CREATE TABLE %s (", table);
+  g_string_printf(query_string, "CREATE TABLE `%s` (`time` timestamp NOT NULL DEFAULT current_timestamp(), ", table);
```
Comment (bazsi) [inline modules/afsql/afsql.c:560]: I think this could be added via configuration. You can specify default values to a field.

**Review by bazsi — COMMENTED** [review]
Apart from the time field which can be added as config this looks good.

**Comment by alltilla** [comment]
I have a vague memory that the quoting might differ in some SQL server kinds (looking at you oracle -_-). If that is the case, we need to do this in a more sophisticated way. There are E2E tests which do not run on these PRs, and they do test the different SQL destinations. They have to be started manually. @HofiOne Sorry for flooding you, but could you kindly help with this also, when you have the time? Thanks! <3

**Comment by HofiOne** [comment]
> I have a vague memory that the quoting might differ in some SQL server kinds (looking at you oracle -_-). If that is the case, we need to do this in a more sophisticated way. There are E2E tests which do not run on these PRs, and they do test the different SQL destinations. They have to be started manually. @HofiOne Sorry for flooding you, but could you kindly help with this also, when you have the time? Thanks! <3

yepp, of course, the main problem now is that we can E2E test only specific branches as the master is still fails, after the last missing fix is merged (#4304) tests will run automatically and we will see the result immediately
the same is true if you wish to pre-test (before merging) these, I can run the tests based on these branches, but first we need a base that will not break the test earlier, far before the test reaches these cases

**Comment by HofiOne** [comment]
@alltilla  Though builds are fine now we have a bunch of issues accessing any 3rd party service, mainly because of outdated credentials, expired licenses, tokens, and accounts, DBO is working on it, I'll keep you updated as there's any progress on this.

**Comment by alltilla** [comment]
> @alltilla Though builds are fine now we have a bunch of issues accessing any 3rd party service, mainly because of outdated credentials, expired licenses, tokens, and accounts, DBO is working on it, I'll keep you updated as there's any progress on this.

Thank you, I really appreciate that! :)

**Comment by bazsi** [comment]
According to this: https://stackoverflow.com/questions/10573922/what-does-the-sql-standard-say-about-usage-of-backtick

using backticks is a MySQLism. Unfortunately libdbi does not offer such quoting functionality.

if you could extract the quoting of identifiers into a separate function, which does it via backticks if the driver is mysql and nothing otherwise, I think this could be merged. Otherwise, this would just break all the other drivers.

**Comment by 143981** [comment]
maybe it would be easier so that it could be written in the config? with 

> table("`${HOST}`")

with this config - table name is empty...

**Comment by bazsi** [comment]
The backtick character is used by the syslog-ng language as well. I am not
sure theres a way to escape it at the moment.

**Comment by bazsi** [comment]
> The backtick character is used by the syslog-ng language as well. I am not sure theres a way to escape it at the moment.

I've just checked, a double backtick would include a single one.

So this should work:

```
table("``${HOST}``")
```

**Comment by 143981** [comment]
> > The backtick character is used by the syslog-ng language as well. I am not sure theres a way to escape it at the moment.
> 
> I've just checked, a double backtick would include a single one.
> 
> So this should work:
> 
> ```
> table("``${HOST}``")
> ```

` version for the right syntax to use near \'.8.164_ `
`query='CREATE TABLE _10.16.8.164_ (time datet....`

no, it do not work..

**Comment by bazsi** [comment]
> > > The backtick character is used by the syslog-ng language as well. I am not sure theres a way to escape it at the moment.
> > 
> > 
> > I've just checked, a double backtick would include a single one.
> > So this should work:
> > ```
> > table("``${HOST}``")
> > ```
> 
> `version for the right syntax to use near \'.8.164_` `query='CREATE TABLE _10.16.8.164_ (time datet....`
> 
> no, it do not work..

the backticks do reach the SQL destination, but then the SQL destination removes invalid characters from the name of
the table and therefore does not allow the backtick character.

Everything is invalid that is not in this set:

```
static gboolean _sql_identifier_is_valid_char(gchar c)
{
  return ((c == '.') ||
          (c == '_') ||
          (c >= '0' && c <= '9') ||
          (g_ascii_tolower(c) >= 'a' && g_ascii_tolower(c) <= 'z'));
}
```

So the only way to do this is via explicit escaping from the SQL module, as you implemented it. Or by replacing the '.' characters with something else before reaching the SQL destination. (e.g. subst() right before hitting the sql() destination)

**Comment by bazsi** [comment]
Are you willing to add the backtick style escaping when formatting the table name?

If you extract the escaping to a function that checks if we are running on MySQL and only performs it there, I'd be willing to merge that.

**Comment by HofiOne** [comment]
Implemented in https://github.com/syslog-ng/syslog-ng/pull/4460

**Pull request closed without merging**